src/main/scala/com/github/agourlay/cornichon/dsl/MacroBasedDsl.scala

```diff
@@ -0,0 +1,41 @@
+package com.github.agourlay.cornichon.dsl
+
+import com.github.agourlay.cornichon.core._
+import com.github.agourlay.cornichon.macros.Macro
+
+import scala.language.experimental.{ macros ⇒ `scalac, please just let me do it!` }
+
+trait MacroBasedDsl {
+  def FeatureMacro(name: String)(body: ⇒ Scenario): FeatureDef = macro Macro.featureImpl
+
+  def ScenarioMacro(name: String)(body: ⇒ Step): Scenario = macro Macro.scenarioImpl
+  def ScenarioMacro(name: String, ignore: Boolean)(body: ⇒ Step): Scenario = macro Macro.scenarioWithIgnoreImpl
```

Comment:
Macro do not play nice with params with default values?
I am fine with the overloaded version, it is just out of curiosity.